packages/renderer/src/lib/compose/ComposeDetailsLogs.svelte

```diff
@@ -0,0 +1,186 @@
+<script lang="ts">
+import { onDestroy, onMount } from 'svelte';
+import { Terminal } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit';
+import 'xterm/css/xterm.css';
+import { TerminalSettings } from '../../../../main/src/plugin/terminal-settings';
+import { getPanelDetailColor } from '../color/color';
+import { isMultiplexedLog } from '../stream/stream-utils';
+import EmptyScreen from '../ui/EmptyScreen.svelte';
+import NoLogIcon from '../ui/NoLogIcon.svelte';
+import type { ComposeInfoUI } from './ComposeInfoUI';
+
+export let compose: ComposeInfoUI;
+
+// Logging element
+let logsXtermDiv: HTMLDivElement;
+let refCompose;
+
+// Log initialization
+let noLogs = true;
+let logsTerminal: Terminal;
+
+$: {
+  if (refCompose && refCompose.status !== compose.status) {
+    logsTerminal?.clear();
+    fetchComposeLogs();
+  }
+  refCompose = compose;
+}
+
+// Terminal resize
+let resizeObserver: ResizeObserver;
+let termFit: FitAddon;
+let currentRouterPath: string;
+
+// Router path for the logging
+let logsRouterPath = `/compose/${encodeURI(compose.name)}/${encodeURI(compose.engineId)}/logs`;
+
+// An array of readable ANSI escape sequence colours against a black terminal background
+// these are the most "readable" colours against a black background
+// No colours like grey, normal blue (cyan instead) or red, since they don't appear very well.
+const ansi256Colors = [
```

Comment:
It looks like pods logs and composeLogs page could share common code instead of copy paste between these two ?

Reply:
you're right, I can move the colourization to another file. 

The other functions I wouldn't be able to move since it ties in quite tightly with iTerm, etc.

I'll refactor this! thanks for pointing it out.